src/api/app/models/project.rb

```diff
@@ -968,6 +976,44 @@ def branch_to_repositories_from(project, pkg_to_enable, opts = {})
     end
   end
 
+  def branch_remote_repositories(project)
```

Comment:
> I would say that it belongs to project... :thinking:

I mean its true that its related to the `Project` model so its logical that the code can go into `project.rb` but one thing I've noticed about this codebase is that there is very little encapsulation of individual components of code which can mean that making changes to code is difficult because there is so much coupling between code which should be isolated from each other. 

So IMO we need to be a bit more careful about how we organize the code, i.e. not just saying "its related to the project so put it in project.rb" or "its related to the source so put it in source_controller.rb". 

This situation is described under the "Tight Coupling" section of this article. The acronyms are bit a silly but I think its a pretty good comparison of various OOP practices:

http://williamdurand.fr/2013/07/30/from-stupid-to-solid-code/

This is just my opinion anyway.. Maybe others prefer to merge this as is?